app/_src/kubernetes-ingress-controller/guides/configure-acl-plugin.md

```diff
@@ -88,251 +79,277 @@ about the algorithms used to construct the JWT, information ("claims") about
+* [Show User JWT](https://jwt.io/#debugger-io?token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ1c2VyLWlzc3VlciJ9.KrdGJ50DYHiElgDULXDl3ZapcHIyILELFoGTIdYB54JTOxIgsaVo6aei2QbYucLK0GCmPNx7ED9J9SfilK_6ZsR7wHq7f7kGozXwgWf3LG5gMak1mzoHnJPi9cwzbSpV-OhJ4q4LkFUi7sGxRPlvSiP2GnxU38rxTBOonJ_y_nkZLXnTlaXanMxCdW_sccP9Y6_Px3_NtoRpVwkf11vj0Dv_9NMqK7WXEj8VPafOMp7NXFSZ0Ebo6vJ89znsA3_XdUqL0_PPZXMZ3ehPVfgz766LTJ8rf1_sQ0W6wxPapXEyiid_eo7JgAtVG4jYHspzuobkmWnJR7lLJOKlm8tVlLQo0e8nOVqV3Ks5e-kqHCfPG-5lD81u5gprG9UB04M8vVw5VN5uT5yHvGWEN2YmA9QsvFjHqTKa7lr0gTXS7lNBCaDLrm2YBSrywZMqs_QOlUtZLTEQSr1B6-cpp_b8gNxNsxzUctca3zj93QJ593qC2ifKxw8XkMab-YhV_lJexjwf1SO_AOBALTIcficI_Z5BVfLWjnEFOl2mEDffDs36fLft-d82tTSYQ76iCgy3tYEqdb3mwPsamaHIcbSYs35xoOocv1kKq81Pq-h6l3aX3yvj9OPfbfO2eS-faC2R0rd6PcjTIzD1v4f48gatnqpp_cKmu0tPfZ3c3dQLP5I&publicKey=-----BEGIN%20PUBLIC%20KEY-----%0AMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAxgqQPxIHPeiVz2Rtq7fF%0Aqria55PKorC6D3s0n3Di5Wp%2F8JtV9u%2BGzcAIviKZxo8urtyu1vBxmxwtDmDkynRS%0A7FpC28O3WqVblVYNEpAQpsTR0fiPnaSRLuJgqTZU3JkHdlkJISCISWjGfN2rqN4f%0AOwKGIiWU0JOWm5HiTG6Uf2S06Fv5OB0rhhlRT2W6hGC43JGnaQ%2F1Ek5zPgiuNfop%0AKjptNFAsgUjWneZFC0toI7ivKudCQWB6v%2FfBn%2F7Lycd%2FqT4DOaYJsE%2Fup23qnsf%2B%0AU8y3emZ8F%2Bs69T5k5aEzIxs89HD7zElKNHCSlUIl%2BGar0h1HTN1QkzuGwppgGrYI%0AH1nkAGYMjhSJa8TvYDI%2FEze4KhYwjfNEnRbuYB74LyGl5Z2imJYlOPb2bYTAtmw0%0A2GNDj27LGFbfhPA4rUE8EadIsE4i4AIdl8UH3OnAYsyj2%2FUbr%2BZ7TtTZYsRlv4MR%0AFVKjO71wonAD3ssZXGz7DKleqvXedEPGXeLbz2BrDKsOGojawLA04yNY0xMT0slN%0Ag%2FKPYWmuug0oONPZhKYsj%2FH0Jk1xiCyJK9B6ItfbAK021NWXrAPHbgUNEbNFy8yS%0A7cPgS3OgEoCMBITKVpJlxvEwuajrewBRUcgS9opz%2BJ94dGd8bnmLRlXK6290Ni6S%0AhF9M5348YA%2F8VIjYq7XBw9sCAwEAAQ%3D%3D%0A-----END%20PUBLIC%20KEY-----%0A)
+
+The `iss` field in the payload matches the value you provided in `--from-literal=key=` when creating the Kubernetes secret.
+
+1. Copy the "Encoded" value and store it in an environment variable for both the `ADMIN_JWT` and `USER_JWT`:
+
+    ```bash
+    export ADMIN_JWT=eyJhbG...
+    export USER_JWT=eyJhbG...
+    ```
+
+1. Send a request with the `Authorization` header.
+
+    ```bash
+    curl -I -H "Authorization: Bearer ${USER_JWT}" http://kong.example/lemon --connect-to kong.example:80:$PROXY_IP
```

Comment:
Why does this use `--connect-to` rather than `Host: kong.example`?

Reply:
I get this error: rajakavitha@198 ~ % curl -I -H "Host:kong.example:80" $PROXY_IP/lemon                                    
HTTP/1.1 401 Unauthorized
Date: Wed, 04 Oct 2023 06:09:58 GMT
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Content-Length: 26
X-Kong-Response-Latency: 3
Server: kong/3.3.1

rajakavitha@198 ~ % echo $USER_JWT
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ1c2VyLWlzc3VlciJ9.KrdGJ50DYHiElgDULXDl3ZapcHIyILELFoGTIdYB54JTOxIgsaVo6aei2QbYucLK0GCmPNx7ED9J9SfilK_6ZsR7wHq7f7kGozXwgWf3LG5gMak1mzoHnJPi9cwzbSpV-OhJ4q4LkFUi7sGxRPlvSiP2GnxU38rxTBOonJ_y_nkZLXnTlaXanMxCdW_sccP9Y6_Px3_NtoRpVwkf11vj0Dv_9NMqK7WXEj8VPafOMp7NXFSZ0Ebo6vJ89znsA3_XdUqL0_PPZXMZ3ehPVfgz766LTJ8rf1_sQ0W6wxPapXEyiid_eo7JgAtVG4jYHspzuobkmWnJR7lLJOKlm8tVlLQo0e8nOVqV3Ks5e-kqHCfPG-5lD81u5gprG9UB04M8vVw5VN5uT5yHvGWEN2YmA9QsvFjHqTKa7lr0gTXS7lNBCaDLrm2YBSrywZMqs_QOlUtZLTEQSr1B6-cpp_b8gNxNsxzUctca3zj93QJ593qC2ifKxw8XkMab-YhV_lJexjwf1SO_AOBALTIcficI_Z5BVfLWjnEFOl2mEDffDs36fLft-d82tTSYQ76iCgy3tYEqdb3mwPsamaHIcbSYs35xoOocv1kKq81Pq-h6l3aX3yvj9OPfbfO2eS-faC2R0rd6PcjTIzD1v4f48gatnqpp_cKmu0tPfZ3c3dQLP5I
rajakavitha@198 ~ % echo $ADMIN_JWT
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJhZG1pbi1pc3N1ZXIifQ.kiHOLThl1SggXGxNhfDuSqP70uf4V80HUoeVBt7S-YqMz0DEL6IUeafMl_LzZyLYQO0qhhNzKL10tBN0krSe3IXsS9t4tlG9kWScDT7BavDyLRAffQ-ylT-PFNk0aXNtQ-0PEN4CVu71j7t7XZcV1USAFpLlr_QD3ruvlnOA2EP_5Cw7Ub9LnGOI_TYbkmYt-OsGXfFsuofytOYeG2_CDMCSNyJC5kNyehoDrFv8JZw4MLwPbOsCbvWo6y0TKN0ydVYyHCc5eu1eqOvdnQlBmL8uBmgBm_9MAyixhbNCum3nRoevb6ySGkCZGwpvEL7vomJcLOSM1naG8OhwNguHiBHt_WkS2FrlACd11xUkFkrtoH7UMpXIgiFOBihHWfEUBfzZcIcrgmsF0rtAo64JXs-km1gwKyIfwb29XQEEWd2-KzEDa2zS7IRQXD9i1U9tGpG7q3PJZLHFGkSuKs4i5mQ8gTPr-lB7X45Yp6g-xqLSA5EvgIo78AJ-G2gJW221VZM5gOvErEmLtDjT9Kk9Cqp-L4vXpWTpAo3sJN2kaQU24O6L8ujDn2XUnNkiU7QDPOQ9TPl5lDhw0VaFtMbGkBJORc2p-3-FTQfV80X6y9RHdyrb3UOhm1rjum2DsOUaacFJPhNcVsDYKTo1bhKoqVDtMYUvNjWP6hawb5H4jyg
rajakavitha@198 ~ % curl -I -H "Authorization: Bearer ${USER_JWT}" http://kong.example/lemon --resolve kong.example:80:$PROXY_IP

curl: (49) Couldn't parse CURLOPT_RESOLVE entry 'kong.example:80:198.19.249.2:80'
rajakavitha@198 ~ % curl -I -H "Authorization: Bearer ${USER_JWT}" http://$PROXY_IP/lemon                           
HTTP/1.1 404 Not Found
Date: Wed, 04 Oct 2023 06:13:56 GMT
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Content-Length: 52
X-Kong-Response-Latency: 1
Server: kong/3.3.1

rajakavitha@198 ~ % curl -I -H "Authorization: Bearer ${USER_JWT}" http://kong.example/lemon --resolve kong.example:80:$PROXY_IP

curl: (49) Couldn't parse CURLOPT_RESOLVE entry 'kong.example:80:198.19.249.2:80'
rajakavitha@198 ~ % curl -I -H "Authorization: Bearer ${USER_JWT}" http://kong.example/lemon --connect-to kong.example:80:$PROXY_IP

HTTP/1.1 200 OK
Content-Type: text/plain; charset=utf-8
Content-Length: 137
Connection: keep-alive
Date: Wed, 04 Oct 2023 06:17:43 GMT
X-Kong-Upstream-Latency: 4
X-Kong-Proxy-Latency: 8
Via: kong/3.3.1

rajakavitha@198 ~ % echo "
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: admin-acl
plugin: acl
config:
  allow: ['admin']
" | kubectl apply -f -

kongplugin.configuration.konghq.com/admin-acl created
rajakavitha@198 ~ % echo "
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: anyone-acl
plugin: acl
config:
  allow: ['admin','user']
" | kubectl apply -f -

kongplugin.configuration.konghq.com/anyone-acl created
rajakavitha@198 ~ % kubectl annotate ingress lemon konghq.com/plugins=admin-acl
kubectl annotate ingress lime konghq.com/plugins=anyone-acl

ingress.networking.k8s.io/lemon annotated
ingress.networking.k8s.io/lime annotated
rajakavitha@198 ~ % kubectl create secret \
  generic admin-acl \
  --from-literal=kongCredType=acl  \
  --from-literal=group=admin

kubectl create secret \
  generic user-acl \
  --from-literal=kongCredType=acl  \
  --from-literal=group=user

secret/admin-acl created
secret/user-acl created
rajakavitha@198 ~ % kubectl patch --type json kongconsumer admin \
  -p='[{
    "op":"add",
    "path":"/credentials/-",
    "value":"admin-acl"
  }]'
kubectl patch --type json kongconsumer user \
  -p='[{
    "op":"add",
    "path":"/credentials/-",
    "value":"user-acl"
  }]'

kongconsumer.configuration.konghq.com/admin patched
kongconsumer.configuration.konghq.com/user patched
rajakavitha@198 ~ % curl -sI http://kong.example/lemon -H "Authorization: Bearer ${ADMIN_JWT}" --resolve kong.example:80:$PROXY_IP | grep HTTP
curl -sI http://kong.example/lime -H "Authorization: Bearer ${ADMIN_JWT}" --resolve kong.example:80:$PROXY_IP | grep HTTP

rajakavitha@198 ~ % curl -sI http://kong.example/lemon -H "Authorization: Bearer ${ADMIN_JWT}" --connect-to kong.example:80:$PROXY_IP | grep HTTP
curl -sI http://kong.example/lime -H "Authorization: Bearer ${ADMIN_JWT}" --connect-to kong.example:80:$PROXY_IP | grep HTTP

HTTP/1.1 200 OK
HTTP/1.1 200 OK
rajakavitha@198 ~ % curl -sI http://kong.example/lemon -H "Authorization: Bearer ${USER_JWT}" --connect-to kong.example:80:$PROXY_IP | grep HTTP
curl -sI http://kong.example/lime -H "Authorization: Bearer ${USER_JWT}" --connect-to kong.example:80:$PROXY_IP | grep HTTP

HTTP/1.1 403 Forbidden
HTTP/1.1 200 OK
rajakavitha@198 ~ %

Reply:
I think this is unrelated. I fixed an issue around the `401` issue. Please update to use `Host` and if there's still a `401` I will investigate

Reply:
Fixed!!